superset-frontend/src/dashboard/components/nativeFilters/FilterBar/index.tsx

```diff
@@ -238,6 +238,9 @@ const FilterBar: React.FC<FiltersBarProps> = ({
         const currentTargets = currentFilter.targets;
         const currentDataMask = currentFilter.defaultDataMask;
         const previousFilter = previousFilters?.[currentFilter.id];
+        if (!previousFilter) {
```

Comment:
Since this is a pre-condition would you mind moving it at the top of the loop just to make the code more readable. Thank you!